packages/mui-system/src/cssVars/createCssVarsProvider.js

```diff
@@ -242,7 +242,21 @@ export default function createCssVarsProvider(options) {
         lightColorScheme,
         mode,
         setColorScheme,
-        setMode,
+        setMode:
+          process.env.NODE_ENV === 'production'
+            ? setMode
+            : (newMode) => {
+                if (theme.colorSchemeSelector === 'media') {
+                  console.error(
+                    [
+                      'MUI: The `setMode` function has no effect if `colorSchemeSelector` is `media` (`media` is the default value).',
+                      'To toggle the mode manually, please configure `colorSchemeSelector` to use a class or data attribute.',
+                      'To learn more, visit https://mui.com/material-ui/customization/css-theme-variables/configuration/#toggling-dark-mode-manually',
```

Comment:
How about we enforce all of those links to be permalink? https://github.com/mui/mui-public/issues/222

```suggestion
                      'To learn more, visit https://mui.com/r/set-mode-media',

```

Especially a hash link, it seems too brittle.